### ISSUE TITLE
File not found error message for gtdbtk.pplacer.scratch

### ISSUE THREAD

**Issue by jserrana** [Ecogenomics/GTDBTk#402]
Hi GTDBTk dev team,

I am running the classify_wf using the newest version 2.1.0/R207_v2. Installation and dependencies are okay (after check_install). I kept getting this error message, `EXCEPTION: FileNotFoundError MESSAGE: [Errno 2] No such file or directory: '/home/jserrana/scratch/gtdbk-scratch/gtdbtk.pplacer.scratch'` when I run the command:

```
gtdbtk classify_wf --extension fa --genome_dir /home/jserrana/projects/sample/ --out_dir /home/jserrana/projects/gtdbtk-sample-out --cpus 60 --scratch_dir /home/jserrana/scratch/gtdbk-scratch --pplacer_cpus 1
```

The task where the error occurred is after: `TASK: Placing 1 bacterial genomes into class-level reference tree 2 (1/2) with pplacer using 1 CPUs (be patient).` I noticed that the `gtdbtk.pplacer.scratch` file (the missing file) from the scratch directory is in that directory during the run, but was deleted after. I tried using different locations for the `--scratch_dir` but got the same error. Could this be related to issue #400?

Thanks a lot for your work and would really appreciate your help on this.

## Environment
- [ ] Installed via pip (include the output of `pip list`)
- [x] Using a conda environment (include the output of `conda list && conda list --revisions`)
- [ ] Using a Docker container (include the `IMAGE ID` of the container)

## Server information
- CPU: `AMD EPYC 7532 32-Core Processor 64`
- RAM: `263572476 kB`
- OS: `cat /etc/os-release`

> NAME="Rocky Linux"
VERSION="8.5 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.5"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.5 (Green Obsidian)"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky Linux"
ROCKY_SUPPORT_PRODUCT_VERSION="8"

## Debugging information
- [x] `gtdbtk.log` has been included: [gtdbtk.log](https://github.com/Ecogenomics/GTDBTk/files/8746228/gtdbtk.log)
- [x] Genomes have been included: [sample.zip](https://github.com/Ecogenomics/GTDBTk/files/8746220/sample.zip)

## Additional comments

I've tried `--cpus 1` (and other values) and  `--force` producing the same error.

**Comment by pchaumeil** [comment]
Hello, 

Unfortunately, I am unable to reproduce your error with the `sample.zip` you have attached. 
GTDB-Tk returns this error because, it seems, pplacer is returning an error when placing the bacterial genome into the class-level reference tree. With pplacer stopping in the early stage, it never creates the scratch file and therefore, when Tk tries to delete it, it returns an error (the file doesn't exist). 
you could have a look at the pplacer log files (`pplacer.class_level.bac120.out`) to make sure pplacer finishes without any error. Those logs can be found under `classify/intermediates/pplacer/tree_<idx>/`

Cheers,
Pierre

**Comment by jserrana** [comment]
Hi Pierre,

Thanks for checking it out. I tried running it with an HPC, and on another local server, which is both using a network filesystem. I've resolved the issue by running the command in the home directory, but since I don't have enough storage, scratch is still on a different disk, but it worked well without errors! Has this something to do with how pplacer does atomic writing to pick a backend since it's failing for network filesystems?

Thanks for your assistance and for managing this amazing tool!

**Comment by pchaumeil** [comment]
_"Has this something to do with how pplacer does atomic writing to pick a backend since it's failing for network filesystems?"_
Sorry I am not too sure about this, I will need to look a bit more in this issue.

**Comment by pchaumeil** [comment]
I could not find any information about this, I am closing the ticket for now

**Comment by nick-youngblut** [comment]
@jserrana did you ever figure out what the error is? I'm using GTDB-Tk v0.3.0 in order to use the r89 database. If I had to guess the problem is with pplacer, which likely crashed before creating the `gtdbtk.pplacer.scratch` file. I'm wondering if the issue is due to a compatibility issue with numpy.

**Comment by nick-youngblut** [comment]
@pchaumeil the dependency versions are not constrained in the [GTDB-Tk v0.3.0 bioconda recipe](https://github.com/bioconda/bioconda-recipes/blob/9b64fbfa7cc3008cb7fa80e2dd79b764e5850c18/recipes/gtdbtk/0.3.0/meta.yaml). So, using this "old" recipe is likely installing a dependency with a version that is "too new" for GTDB-Tk v0.3.0, so essentially GTDB-Tk v0.3.0 cannot be used for legacy workflows (the problem with leaving dependency versions open-ended).

Do you (or other GTDB-Tk) developers know which versions of all GTDB-TK dependencies actually work with v0.3.0?

**Comment by aaronmussig** [comment]
Hi Nick,

I agree that leaving dependencies open-ended can create challenges in some situations. At the same time, it is important to ensure that GTDB-Tk can be easily incorporated into other conda environments that may share dependencies. While I think conda as a whole is good, this is certainly its biggest pitfall that I can't see being resolved any time soon. 

Conda rant aside... I think your best bet here is to run GTDB-Tk in a Docker container that was built at the time of release, you can find them here: https://quay.io/repository/biocontainers/gtdbtk?tab=tags

Additionally, I've also pulled together the dependencies from our cluster for GTDB-Tk 0.3.3 (sorry, for some reason 0.3.0 isn't there). 

```
name: gtdbtk-0.3.3
channels:
  - conda-forge
  - bioconda
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - boost=1.70.0=py27h9de70de_1
  - boost-cpp=1.70.0=h8e57a91_2
  - bzip2=1.0.8=h516909a_1
  - ca-certificates=2019.11.28=hecc5488_0
  - certifi=2019.11.28=py27_0
  - dendropy=4.4.0=py_1
  - fastani=1.2=he1c1bb9_0
  - fasttree=2.1.10=h14c3975_3
  - future=0.18.2=py27_0
  - gtdbtk=0.3.3=py27_0
  - hmmer=3.2.1=he1b5a44_2
  - icu=64.2=he1b5a44_1
  - libblas=3.8.0=14_openblas
  - libcblas=3.8.0=14_openblas
  - libffi=3.2.1=he1b5a44_1006
  - libgcc-ng=9.2.0=hdf63c60_0
  - libgfortran-ng=7.3.0=hdf63c60_2
  - liblapack=3.8.0=14_openblas
  - libopenblas=0.3.7=h5ec1e0e_4
  - libstdcxx-ng=9.2.0=hdf63c60_0
  - ncurses=6.1=hf484d3e_1002
  - numpy=1.16.5=py27h95a1406_0
  - openssl=1.1.1d=h516909a_0
  - pip=19.3.1=py27_0
  - pplacer=1.1.alpha19=1
  - prodigal=2.6.3=h516909a_2
  - python=2.7.15=h5a48372_1009
  - readline=8.0=hf8c457e_0
  - setuptools=42.0.2=py27_0
  - sqlite=3.30.1=hcee41ef_0
  - tk=8.6.10=hed695b0_0
  - wheel=0.33.6=py27_0
  - xz=5.2.4=h14c3975_1001
  - zlib=1.2.11=h516909a_1006
prefix: /srv/sw/miniconda3/envs/gtdbtk-0.3.3
```

**Comment by nick-youngblut** [comment]
I've contributed many recipes to bioconda, and I can admit that I've left some dependencies open-ended. Maybe the bioconda core team should implement rules about designating specific versions for dependencies, but then there's the issue of listing *all* relevant dependencies. 

> I think your best bet here is to run GTDB-Tk in a Docker container that was built at the time of releas

I did try that yesterday and received the same error, so maybe it isn't a dependency conflict? 🤷 

Thanks for putting together the conda env for 0.3.3!! v0.3.3 is too new for Release89, correct? I guess that I'll have to use r95.
